app/controllers/api/v2/registration_commands_controller.rb

```diff
@@ -18,6 +18,9 @@ class RegistrationCommandsController < V2::BaseController
         param :insecure, :bool, desc: N_("Enable insecure argument for the initial curl")
         param :packages, String, desc: N_("Packages to install on the host when registered. Can be set by `host_packages` parameter, example: `pkg1 pkg2`")
         param :update_packages, :bool, desc: N_("Update all packages on the host")
+        param :repo, String, desc: N_("Repository URL / details, for example for Debian OS family: 'deb http://deb.example.com/ buster 1.0', for Red Hat and SUSE OS family: 'http://yum.theforeman.org/client/latest/el8/x86_64/'"), deprecated: true
```

Comment:
When I run `bin/hammer host-registration generate-command --help`, I don't see any information about the deprecation.
@ofedoren What does the `deprecated` do? I thought it would mark the fields in `hammer --help` output as deprecated.

Reply:
Unfortunately, hammer doesn't do that. Mostly we need to do that manually, e.g. in the command definition itself we need to add `build_options without: repo, :repo_gpg_key_url`. And then add them as options manually with the deprecation warning like in https://github.com/theforeman/hammer-cli-foreman/blob/master/lib/hammer_cli_foreman/command_extensions/compute_resource_subcommand.rb#L5

Reply:
So how is the workflow? Shall I open a follow-up PR in parallel or do we get this one in, first?

Reply:
So how do we proceed here? I think it should go in asap because https://github.com/theforeman/foreman/pull/10162 has a breaking change in the API (that is reverted here) and is part of 3.11. So the fix should also go into 3.11